### PR TITLE
Add translation support

### LINKED ISSUE #12: Enable translations
While we won't be able to translate the contents of the job listings, we can enable translations of the app itself.

### PR DESCRIPTION
**Pull request by maroun-baydoun**
I chose not to use a third-party library for the time being. Instead, I re-used some code I had created for a different project to give basic support for multiple language. The API is similar to `react-intl`. In the future, we could easily replace the code with a third-party solution if there's need for that. 

I added only support for English with a single string to test the flow. We will add more languages and strings as we decide on what languages to support and the content to show on the web app. 

Closes #12

### CODE REVIEW

File: src/i18n/useIntl.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+type formatMessageArgs = {
+  id: string;
+  values?: Record<string, string>;
+  defaultMessage?: string;
+};
+
+const useIntl = () => {
+  const messages = useContext(IntlContext);
+
+  return useMemo(
```
Comment (lopis) [inline src/i18n/useIntl.ts:14]: Good one. TIL.

File: src/i18n/index.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+
```
Comment (lopis) [inline src/i18n/index.ts]:
```suggestion
```

File: src/i18n/index.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import enTranslationMessages from '../../translations/en.json';
+
+const LANGUAGES = ['en'];
+const DEFAULT_LOCALE = 'en';
+
```
Comment (lopis) [inline src/i18n/index.ts:5]:
```suggestion
```

File: src/i18n/index.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+const translationMessages: { [locale: string]: { [key: string]: string } } = {
+  en: enTranslationMessages as { [key: string]: string },
+};
+
+export const getTranslationMessages = (locale: string) => {
+  return translationMessages[locale] || translationMessages[DEFAULT_LOCALE];
+};
+
```
Comment (lopis) [inline src/i18n/index.ts:13]:
```suggestion
```

File: src/i18n/index.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  if (LANGUAGES.indexOf(browserLang) !== -1) {
+    return browserLang;
+  }
+  if (fallbackLang && LANGUAGES.indexOf(fallbackLang) !== -1) {
+    return fallbackLang;
+  }
+
+  return DEFAULT_LOCALE;
+}
+
```
Comment (lopis) [inline src/i18n/index.ts:26]:
```suggestion
```

File: translations/en.json [hunk 6]
```diff
@@ -0,0 +1,3 @@
+{
+    "publishJobListing": "Publish Job Listing"
+}
```
Comment (lopis) [inline translations/en.json]:
```suggestion
}

```

File: src/i18n/index.ts [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+import enTranslationMessages from '../../translations/en.json';
```
Comment (lopis) [inline src/i18n/index.ts:1]: The main issue here is that we'll be importing all languages always. But we can worry about that in the future. I like this solution!

**Review by lopis — CHANGES_REQUESTED** [review]
I like this solution. It's quite concise. I just added a couple suggestions to missing/unnecessary line breaks.

File: src/i18n/index.ts (hunk 7 above)
Comment (maroun-baydoun) [reply src/i18n/index.ts:1]: You're right. I didn't want to complicate the solution at this stage. We will definitely tackle that once we add more languages.

**Review by lopis — APPROVED** [review]
👍🏼

**Pull request merged**